examples/atari/train_drqn_ale.py

```diff
@@ -0,0 +1,243 @@
+    parser.add_argument('--no-frame-stack', action='store_true', default=False,
+                        help='Disable frame stacking so that the agent can'
+                             ' only see the current screen.')
+    parser.add_argument('--episodic-update-len', type=int, default=10,
+                        help='Maximum length of sequences for updating'
+                             ' recurrent models')
+    parser.add_argument('--batch-size', type=int, default=32,
+                        help='Number of transitions (in a non-recurrent case)'
+                             ' or sequences (in a recurrent case) used for an'
+                             ' update.')
+    args = parser.parse_args()
+
+    import logging
+    logging.basicConfig(level=args.logging_level)
+
+    # Set a random seed used in ChainerRL.
+    misc.set_random_seed(args.seed, gpus=(args.gpu,))
+
+    # Set different random seeds for train and test envs.
+    train_seed = args.seed
+    test_seed = 2 ** 31 - 1 - args.seed
+
+    args.outdir = experiments.prepare_output_dir(args, args.outdir)
+    print('Output files are saved in {}'.format(args.outdir))
+
+    def make_env(test):
+        # Use different random seeds for train and test envs
+        env_seed = test_seed if test else train_seed
+        env = atari_wrappers.wrap_deepmind(
+            atari_wrappers.make_atari(args.env, max_frames=args.max_frames),
+            episode_life=not test,
+            clip_rewards=not test,
+            flicker=args.flicker,
+            frame_stack=not args.no_frame_stack,
+        )
+        env.seed(int(env_seed))
+        if test:
+            # Randomize actions like epsilon-greedy in evaluation as well
+            env = chainerrl.wrappers.RandomizeAction(env, args.eval_epsilon)
+        if args.monitor:
+            env = gym.wrappers.Monitor(
+                env, args.outdir,
+                mode='evaluation' if test else 'training')
+        if args.render:
+            env = chainerrl.wrappers.Render(env)
+        return env
+
+    env = make_env(test=False)
+    eval_env = make_env(test=True)
+    print('Observation space', env.observation_space)
+    print('Action space', env.action_space)
+
+    n_actions = env.action_space.n
+    if args.recurrent:
+        # Q-network with LSTM
+        q_func = chainerrl.links.StatelessRecurrentSequential(
+            L.Convolution2D(None, 32, 8, stride=4),
+            F.relu,
+            L.Convolution2D(None, 64, 4, stride=2),
+            F.relu,
+            L.Convolution2D(None, 64, 3, stride=1),
+            functools.partial(F.reshape, shape=(-1, 3136)),
+            F.relu,
+            L.NStepLSTM(1, 3136, 512, 0),
+            L.Linear(None, n_actions),
+            DiscreteActionValue,
+        )
+        # Replay buffer that stores whole episodes
+        rbuf = replay_buffer.EpisodicReplayBuffer(10 ** 6)
+    else:
+        # Q-network without LSTM
+        q_func = chainer.Sequential(
```

Comment:
Any reason not to copy this code: https://github.com/chainer/chainerrl/blob/master/examples/atari/train_dqn_ale.py#L50?

Reply:
Partly because it is consistent with a recurrent version, partly because I think it is easier to understand, thus better as an example.